src/Common/src/Internal/Cryptography/BasicSymmetricCipherBCrypt.cs

```diff
@@ -30,9 +30,9 @@ public BasicSymmetricCipherBCrypt(SafeAlgorithmHandle algorithm, CipherMode ciph
 
             _hKey = algorithm.BCryptImportKey(key);
 
-            if (effectiveKeyLength != 0)
+            if (ownsParentHandle)
             {
-                Cng.SetEffectiveKeyLength(_hKey, effectiveKeyLength);
+                _hKey.SetParentHandle(algorithm);
```

Comment:
This construction makes the key handle being active keep the algorithm handle active, but doesn't cause the algorithm handle to Dispose (it ends up going off to the finalizer).

Honestly we could *always* set the parent handle and that'd be safer (though since we know we static-cache the rest of the algorithm handles it doesn't really matter).

But if we want to avoid finalization we'd need to do something slightly different, like conditionally promoting the algorithm handle to a field on BasicSymmetricCipherBCrypt and disposing it in Dispose.